File: Python Fundamentals/Mid Exam problems/array_modifier.py
sequence = list(map(int, input().split()))

while True:
    command = input()
    if command == 'end':
        print(', '.join(list(map(str, sequence))))
        break

    command_list = command.split()
    if command_list[0] == 'swap':
        element = sequence[int(command_list[1])]
        sequence[int(command_list[1])] = sequence[int(command_list[2])]
        sequence[int(command_list[2])] = element

    elif command_list[0] == 'multiply':
        sequence[int(command_list[1])] *= sequence[int(command_list[2])]

    else:
        sequence = [x - 1 for x in sequence]

File: JavaScript Fundamentals/Mid Exams/Mid Exam 2/arrayModifier.js
function solve(arr) {
    let numbers = arr.shift().split(' ').map(x => Number(x));
    for (let command of arr) {
        if (command == 'end') break;
        else if (command == 'decrease') {
            for (let index = 0; index < numbers.length; index++) {
                numbers[index]--;
            }
        }
        let tokens = command.split(' ');
        if (tokens[0] == 'swap') {
            let a = numbers[Number(tokens[1])];
            let b = numbers[Number(tokens[2])];
            numbers[Number(tokens[1])] = b;
            numbers[Number(tokens[2])] = a;
        } else if (tokens[0] == 'multiply') {
            numbers[Number(tokens[1])] = numbers[Number(tokens[1])] * numbers[Number(tokens[2])];
        }
    }
    console.log(numbers.join(', '));
}
